remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java

```diff
@@ -171,6 +174,9 @@ public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheN
                   if (!knownTypes.contains(typeName)) {
                      throw new CacheConfigurationException("The declared indexed type '" + typeName + "' is not known. Please register its proto schema file first.");
                   }
+                  if (searchMapping == null) {
```

Comment:
```searchMapping``` is null well before this loop, so why not complain earlier about it @gustavonalle ?

Reply:
I put it here so that it can report that a specific type is not indexed. Where do you suggest to move it?

Reply:
a few lines above we have a check:  ```if (cfg.indexing().enabled()) {``` so how can searchMapping be null then here?  it seems the only way (according to SerializationContextSearchMapping class) is if there are not indexed entities configured for a cache that is declared as indexed, and that is something we also catch earlier. So my question is when is this newly introduced CacheConfigurationException (typeNotIndexed) happening, which btw I think duplicates another message?

Reply:
> if (cfg.indexing().enabled()) so how can searchMapping be null then here? 

You can have an odd situation where the cache is indexed, but none of the entities are marked as indexed. This causes some issues, for e.g. https://issues.redhat.com/browse/ISPN-12218
I hope this validation will prevent it

Reply:
> and that is something we also catch earlier

I don't remember seeing this being handled

Reply:
anyway, there's a test for this error reporting https://github.com/infinispan/infinispan/pull/8719/files#diff-b375b40160f198a071885221ebdbb838

If you move the searchmapping check, and the test still passes, good!

Reply:
Looking deeper at this, the search mapping null check is wrong @gustavonalle . See the latest WIP commit I added and the commit comment.

Reply:
And I pushed another WIP, that fixes the issue.  After you have a quick look at it @gustavonalle I think I can squash those and merge this.

Reply:
Good catch @anistor let's wait for CI

Reply:
ci is taking a nap.